tests/unit/s2n_signature_algorithms_test.c

```diff
@@ -1014,7 +1369,7 @@ int main(int argc, char **argv)
             server_conn->security_policy_override = &sha384_policy;
 
             EXPECT_FAILURE_WITH_ERRNO(s2n_negotiate_test_server_and_client(server_conn, client_conn),
-                    S2N_ERR_INVALID_SIGNATURE_ALGORITHM);
+                    S2N_ERR_INVALID_SIGNATURE_SCHEME);
```

Comment:
I think I'm missing this - how is this throwing a different error now?

Reply:
So with the addition of our fix for the SHA1 security issue, s2n_choose_default_sig_scheme could return s2n_null_sig_scheme. From [the code](https://github.com/aws/s2n-tls/blob/main/tls/s2n_signature_algorithms.c#L271C1-L276):
> We cannot bail with an error here because existing logic assumes that this method should always succeed and calls it even when no default is actually necessary. If no valid default exists, set an unusable, invalid empty scheme.

Now s2n_choose_sig_scheme_from_peer_preference_list could probably have safely checked that the signature scheme wasn't s2n_null_sig_scheme and thrown "S2N_ERR_INVALID_SIGNATURE_SCHEME", but it didn't (I'm guessing we were trying to keep the patch small?). So instead the error came when we later tried to use the null scheme to choose a certificate [here](https://github.com/aws/s2n-tls/blob/a9ce5e4ae87a00a8be1b2435ea78b378d8e80765/tls/s2n_auth_selection.c#L73-L75).

Reply:
Makes sense, thanks!